storage/src/test/java/org/apache/kafka/tiered/storage/utils/TieredStorageTestUtils.java

```diff
@@ -55,7 +57,7 @@ public class TieredStorageTestUtils {
 
     // Log cleanup interval is configured to be 500 ms. We need to wait at least that amount of time before
     // segments eligible for deletion gets physically removed.
-    public static final Integer STORAGE_WAIT_TIMEOUT_SEC = 5;
+    public static final Integer STORAGE_WAIT_TIMEOUT_SEC = 10;
```

Comment:
To get the context, Why the timeout was increased?

Reply:
The AlterLogDirTest seems flaky:

```
Build / JDK 17 and Scala 2.13 / executeTieredStorageTest(String).quorum=kraft – org.apache.kafka.tiered.storage.integration.AlterLogDirTest
```

Reply:
Yes, I think the test is flaky because the CI environment is quite slow, and maybe the IO is slower than we thought. From the log I can see, at `08:59:27,187`, we copied `00000000000000000002.log` to remote. And after 1 second of `00000000000000000002.log` copied to remote (`08:59:28,300`), it timed out and all resources started to get closed.

I've waited 10 seconds for the log deletion to get completed, but obviously it's not enough for CI env. I've increased to 20 seconds and see if it fixes the issue. I think we've done what we can do to make it faster (i.e. set the configs to speed up the tests)

```
[2024-05-04 08:59:27,187] INFO [RemoteLogManager=0 partition=DcnVRVRSQd675ZLtCIn21A:topicB-0] Copied 00000000000000000002.log to remote storage with segment-id: RemoteLogSegmentId{topicIdPartition=DcnVRVRSQd675ZLtCIn21A:topicB-0, id=gcVp790dRlmFCr_0tN0NTg} (kafka.log.remote.RemoteLogManager$RLMTask:792)

[2024-05-04 08:59:28,300] INFO Closing topic-based RLMM resources 
[2024-05-04 08:59:28,304] INFO Closing the instance (org.apache.kafka.server.log.remote.metadata.storage.ConsumerTask:328)
[2024-05-04 08:59:28,308] INFO Exited from consumer task thread (org.apache.kafka.server.log.remote.metadata.storage.ConsumerTask:151)
```

Reply:
The test is still flaky. Investigating.